### ISSUE TITLE
cifs IPv6 support

### ISSUE THREAD

**Issue by Vladimir-csp** [IgnorantGuru/udevil#46]
Currently udevil can not use IPv6 URLs:

```$ ping6 ::1
PING ::1(::1) 56 data bytes
64 bytes from ::1: icmp_seq=1 ttl=64 time=0.030 ms
64 bytes from ::1: icmp_seq=2 ttl=64 time=0.051 ms
^C
--- ::1 ping statistics ---
2 packets transmitted, 2 received, 0% packet loss, time 999ms
rtt min/avg/max/mdev = 0.030/0.040/0.051/0.012 ms```

```$ udevil mount  smb://[::1]/public
udevil: trying cifs as guest
mount error: could not resolve address for ::1: Unknown error
udevil: trying with user=localusername
mount error: could not resolve address for ::1: Unknown error```

**Comment by IgnorantGuru** [comment]
Thanks, please test next branch (see BUILD NEXT in README).  Note that "ip=" should **NOT** be added to udevil.conf allowed_options (this may cause a security problem).

This change also causes udevil to pass literal ipv6 hosts straight through for any fstype.  Previously, it stripped the square brackets [ ] from the hostname.  Not sure how other filesystems handle these hostnames or why I stripped them originally (except that it needed to be done to resolve the IP).  This may require testing/adjustment on a per-filesystem basis.

**Comment by Vladimir-csp** [comment]
Sorry for late testing, here are the results:
```
$ LANG=C udevil mount smb://[::1]/public
udevil: trying cifs as guest
Mounted //::1/public at /media/smb-::1-public

$ cat /proc/mounts
...
//::1/public /media/smb-::1-public cifs rw,nosuid,nodev,noexec,relatime,vers=1.0,cache=strict,domain=MINDFLASH4,uid=1000,noforceuid,gid=100,noforcegid,addr=0000:0000:0000:0000:0000:0000:0000:0001,file_mode=0755,dir_mode=0755,nounix,serverino,rsize=61440,wsize=65536,actimeo=1 0 0
...

$ LANG=C udevil umount /media/smb-\:\:1-public/
udevil: error 36: lookup host '' failed

$ LANG=C udevil umount smb://[::1]/public
```
So, basically it works, but there are two problems: ':' characters in mountpoint path and inability to unmount by mountpoint path (these are not connected, same happens with custom mountpoint without ':'.

**Comment by IgnorantGuru** [comment]
Thanks for testing.  Why are colons (:) in the mountpoint path a problem?  Do you suggest stripping or replacing them with something else?  Suggestions?  jfyi, you don't need to escape colons:

    echo /media/smb-::1-public
        /media/smb-::1-public

I'll take a look at error 36.

**Comment by IgnorantGuru** [comment]
The reason for error 36 above is that the URL which cifs puts in mtab:

    //::1/public

contains a literal ipv6 host but without the square brackets [ ] around the host address.  I don't think that url format is standard and udevil requires the brackets when parsing such a url.  (Otherwise if a port is specified, as in //HOST:PORT/SHARE, things get confusing or impossible to parse correctly.)  However, since cifs accepts the port as an option rather than in the url, it's not a problem for them.

I have added a fix in 6baf170 which corrects this url before udevil parses it, adding the brackets if needed.  This fix is activated if the mountpoint path's associated url begins with "//" and contains a colon without a preceding square bracket.  This should correct your issue (I didn't do a live test with cifs though, I just faked /proc/mounts and it got past the lookup error).  Is this working for you?

You might want to also try passing a port in the udevil url to make sure that's all working as expected:

    udevil mount smb://[::1]:445/public
